source/patterns/@aws-solutions-constructs/core/lib/s3-bucket-helper.ts

```diff
@@ -144,53 +144,38 @@ function s3BucketWithLogging(scope: Construct,
   }];
 
   // Create the Application Bucket
-  let bucketprops: s3.BucketProps;
+  let bucketProps: s3.BucketProps;
   let loggingBucket;
   const _bucketId = bucketId ? bucketId + 'S3Bucket' : 'S3Bucket';
   const _loggingBucketId = bucketId ? bucketId + 'S3LoggingBucket' : 'S3LoggingBucket';
 
-  if (s3BucketProps?.serverAccessLogsBucket) {
-    // Attach the Default Life Cycle policy ONLY IF the versioning is ENABLED
-    if (s3BucketProps.versioned === undefined || s3BucketProps.versioned) {
-      bucketprops = DefaultS3Props(undefined, lifecycleRules);
-    } else {
-      bucketprops = DefaultS3Props();
-    }
-  } else if (logS3AccessLogs !== false) {
+  if (logS3AccessLogs !== false && !userBucketProps?.serverAccessLogsBucket) {
```

Comment:
I think that creating the additional functions is unnecessary as it performs the same functionality.  I believe we should take advantage of the built in javascript operators for its purpose. I have changed `!userBucketProps?.serverAccessLogsBucket` to `!(props.bucketProps?.serverAccessLogsBucket)` for clarity. I left it as `props.logS3AccessLogs !== false` because the real default is undefined allowing it to pass for use cases when it is undefined or true. Will also add a comment above the line. Let me know what you think